modules/gateway/peers.go

```diff
@@ -26,10 +24,37 @@ const (
 	minNodeListLen = 100
 )
 
+var (
+	// The gateway will sleep this long between incoming connections.
+	acceptInterval = func() time.Duration {
```

Comment:
The gateway has a tight loop where it accepts an incoming connection and then spawns a goroutine to handle it. To prevent the loop from running too quickly (an attack vector), we sleep for `acceptInterval` before accepting another connection.